errors/errors.go

```diff
@@ -67,7 +68,8 @@ func New(errType ErrorType, msg string, args ...interface{}) error {
 
 // Is is a convenience function for testing the internal type of an BoulderError
 func Is(err error, errType ErrorType) bool {
-	bErr, ok := err.(*BoulderError)
+	var bErr *BoulderError
+	ok := errors.As(err, &bErr)
```

Comment:
In general, my feedback has been to keep local style as much as possible when doing these changes. In this case, since the code will look stilted after the change (why a separate `if` and then a comparison?) I recommend combining things to:

```
return errors.As(err, &bErr) && bErr.Type == errType
```